File: tests/fixtures/openai-cua/openai-cua-no-domain-allowlist-fixed/hostname-check-before-click.ts
const ALLOWED_DOMAINS = ['example.com', 'app.example.com'];

export async function executeComputerAction(page: any, action: { type: string; x: number; y: number }) {
  const host = new URL(page.url()).hostname;
  if (!ALLOWED_DOMAINS.includes(host)) {
    throw new Error(`Blocked action on disallowed domain: ${host}`);
  }

  if (action.type === 'click') {
    await page.mouse.click(action.x, action.y);
  }
}

File: tests/fixtures/openai-cua/openai-cua-no-domain-allowlist-fixed/hostname_check_before_click.py
from urllib.parse import urlparse

ALLOWED_DOMAINS = {"example.com", "app.example.com"}


def handle_action(page, action):
    hostname = urlparse(page.url).hostname
    if hostname not in ALLOWED_DOMAINS:
        raise ValueError(f"Blocked action on untrusted domain: {hostname}")
    if action["type"] == "click":
        page.click(action["x"], action["y"])
